### DISCUSSION TITLE
Uninstall on master take down etcd quorun

### DISCUSSION THREAD

**Discussion by giovannicandido** [rancher/rke2#2002] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

When uninstalling rke from a cluster with a quorum the cluster is left in unusable state where nothing works anymore.

Suppose we have a cluster with 3 masters. You clean up two of them with the uninstall script. The master that is left is not usable anymore, the entire cluster is lost.

**Describe the solution you'd like**

When uninstalling one master, I would like that the rke fist drain the cluster and remove it from etcd.

**Describe alternatives you've considered**

If that is not possible, I accept a warning messaging in the uninstallation as a good solution for this, let the administrator know it need to drain the node and that it needs to take care with the quorum of etcd to not lost its cluster.

**Additional context**

Maybe a warning from a worker node is also a good improvement?

**Comment by brandond** [comment, marked as answer]
You need to delete the node from the cluster (`kubectl delete node`) prior to uninstalling it so that it is removed from the etcd cluster. Otherwise, by uninstalling you've just turned the node off without reducing the cluster membership, which will (as you noted) break quorum.

Also note that you can always use the `--cluster-reset` flag on the surviving node to reset the cluster down to a single member without losing any data.
